src/Metadata/AssemblyMetadata.Methods.cs

```diff
@@ -83,7 +83,7 @@ public BlobHandle GetMethodOrConstructorSignature(MethodBase methodBase)
                     {
                         if (mi != null)
                         {
-                            retEnc.FromSystemType(mi.ReturnType, this);
+                            retEnc.FromSystemType(((MethodInfo)methodBase).ReturnType, this);
```

Comment:
It took me a while to understand what's going on here. Nice catch! I would suggest to avoid re-using the `methodBase` variable, and introduce a new one altogether for the sake of clarity. Then, in the same file, rather than writing `mi != null`, I would rather suggest to write `methodBase is MethodInfo` which makes the validity of the cast more locally obvious.

Reply:
No prob... I'll revisit and clean up soon.